### ISSUE TITLE
mount fails to open 0 byte files in rw mode properly

### ISSUE THREAD

**Issue by bittaurus** [ncw/rclone#1581]
I have mounted a google drive remote, with and without crypt.  Both of them exhibit the same problem.  Readwrite file open on a 0 byte file returns failure.  It's not a permissions issue, I can open 1+ byte files just fine.

rclone v1.37-064-g38dabcf6β
- os/arch: linux/amd64
- go version: go1.8.3

rclone mount

```
2017/08/05 04:48:55 DEBUG : rclone: Version "v1.37-064-g38dabcf6β" starting with parameters ["rclone.beta" "-vv" "mount" "crupt:" "pdx"]
2017/08/05 04:48:57 INFO  : Encrypted drive 'crupt:': Modify window is 1ms
2017/08/05 04:48:57 DEBUG : Encrypted drive 'crupt:': Mounting on "pdx"
2017/08/05 04:48:57 DEBUG : : Root:
2017/08/05 04:48:57 DEBUG : : >Root: node=/, err=<nil>
...
2017/08/05 04:52:44 DEBUG : Directory/: Lookup: name="0bytes.txt"
2017/08/05 04:52:44 DEBUG : Directory/: >Lookup: node=Directory/0bytes.txt, err=<nil>
2017/08/05 04:52:44 DEBUG : Directory/0bytes.txt: Attr:
2017/08/05 04:52:44 DEBUG : Directory/0bytes.txt: >Attr: a=valid=1m0s ino=0 size=0 mode=-rw-r--r--, err=<nil>
2017/08/05 04:52:44 DEBUG : Directory/0bytes.txt: Open: flags=OpenReadWrite+0x40000
2017/08/05 04:52:44 DEBUG : Directory/0bytes.txt: >Open: fh=<nil>, err=can't open for read and write simultaneously
```

**Comment by ncw** [comment]
I think you will get this

> 2017/08/05 04:52:44 DEBUG : Directory/0bytes.txt: >Open: fh=<nil>, err=can't open for read and write simultaneously

On any size file if you open it for read and write?

In my tests I do anyway.

I hope to fix this as part of #711

**Comment by bittaurus** [comment]
Thanks, rw access is probably not supported by the gcloud API, (but I could be wrong).  I'm taking the issue to libtorrent, which is failing when it can't open a file in rw when it just needs to read it.
